TAO/tao/Storable_FlatFileStream.h

```diff
@@ -84,6 +96,37 @@ namespace TAO
     virtual Storable_Base& operator << (ACE_INT64 );
     virtual Storable_Base& operator >> (ACE_INT64 &);
 
+#if defined (ACE_HAS_CPP11)
+    // Avoid duplication for the underlying type of size_t
+    template <typename Dummy = Storable_Base &>
+    typename std::enable_if<std::is_same<Dummy, Storable_Base &>::value &&
+                            !std::is_same<ACE_UINT64, size_t>::value &&
```

Comment:
What when the file is created on a 64bit system and after that read on a 32bit system, wouldn’t that be a problem when we add support for size_t?

Reply:
I'm not sure I understand the question. But the operators for `size_t` are guaranteed to be defined only when the underlying type of `size_t` is different than `ACE_UINT64` and `ACE_UINT32`.

Reply:
My question is whether we should allow the API usage using size_t, at that moment we marshal something with a different size what happens when the file is read on a different platform? Shouldn't the caller use UINT32/64 explicitly, that way the size on send/receive is known to be of a fixed size.

Reply:
I think I understand the question and the potential problem with `size_t` now. Do you have any suggestions for this? I guess a simple way to fix is to use `ACE_UINT64_FORMAT_SPECIFIER_ASCII` in both output and input operators for `size_t`. That way we can avoid a potential bug when `size_t` is 64-bit on one platform and 32-bit on the other.

Reply:
Yes, let the caller cast before writing/reading, he should use a fixed size type when using any CORBA related API

Reply:
This is fixed now.